plugins-extra/query_enhancements/public/plugin.tsx

```diff
@@ -82,6 +91,26 @@ export class QueryEnhancementsPlugin
       },
     });
 
+    data.__enhance({
+      ui: {
+        query: {
+          language: 'SQL Async',
```

Comment:
we can probably do this based on data frame type.

we could even consider keeping this language here but then making a new ui enhancement that just makes it a functional language for now then the data source will automatically set if it it should use the async sql or not

would you mind looking into that? don't hesitate to ask questions.